bin/main.ml

```diff
@@ -426,6 +426,32 @@ let runtest =
           $ Arg.(value & pos_all string ["."] name_))
   , Term.info "runtest" ~doc ~man)
 
+let clean =
+  let doc = "Clean the project." in
+  let man =
+    [ `S "DESCRIPTION"
+    ; `P {|Removes files added by jbuilder such as _build, .install, and .merlin|}
+    ; `Blocks help_secs
+    ]
+  in
+  let go common =
+    set_common common ~targets:[];
+    Build_system.load_trace ()
+    |> function
+    | None ->
+      Format.eprintf
+        "@{<error>Error@}: Missing trace file: _build/.db\n\
+         Hint: Is the _build directory corrupt? Try rebuilding? @."
+    | Some trace ->
+      begin
+        Hashtbl.iter trace ~f:(fun ~key ~data:(_) -> Path.unlink key);
```

Comment:
Some of the files might not exist anymore, for instance if the user renamed a directory there will still be an entry for the old .merlin file. We need to use `Path.unlink_no_error`.